### PR TITLE
Assignments, peerreviews, mock submissions 

### PR DESCRIPTION
**Pull request by WilliamWallace1272**
- can create, edit, and get peer reviews
- can create and edit assignments
- can mock submissions for an essay (right now does for all students assigned the assignment)

### CODE REVIEW

File: api/index.php [hunk 1]
```diff
@@ -125,6 +127,98 @@ function decode_json_throw_errors($inputString) {
     return $response;
 });
 
+####################### ASSIGNMENTS ###########################3
+$app->get('/assignments/get/all', function (Request $request, Response $response) use ($dataMgr){
+
```
Comment (madhavsuresh) [inline api/index.php]: whitespace

File: api/index.php [hunk 2]
```diff
@@ -125,6 +127,98 @@ function decode_json_throw_errors($inputString) {
...
+$app->post('/assignment/update', function( Request $request, Response $response) use ($dataMgr){
+    $json_params = $request->getBody();
+    $params = json_decode($json_params,true); 
+    $dataMgr->setCourseFromID(new CourseID($params['courseID']));
+    $assignment = createAssignment($params);
+    
+});
+
+
```
Comment (madhavsuresh) [inline api/index.php]: whitespace

File: api/index.php [hunk 3]
```diff
@@ -125,6 +127,98 @@ function decode_json_throw_errors($inputString) {
...
+$app->post('/rubric/create',function(Request $request, Response $response) use ($dataMgr){
+    
+    $params = json_decode($request->getBody(),true);
+    $dataMgr->setCourseFromID(new CourseID($params['courseID']));
+    #actually don't think you even need the course name, but not 100% about the whole mta system so its here
+    unset($params['courseID']); #get rid of it cause not needed for the rubric
+    $assignment = $dataMgr->getAssignment(new AssignmentID($params['assignmentID'])); 
+    setup_radio_question($assignment, $params);
+
```
Comment (madhavsuresh) [inline api/index.php]: why???

File: api/index.php [hunk 4]
```diff
@@ -125,6 +127,98 @@ function decode_json_throw_errors($inputString) {
...
+    $params = json_decode($request->getBody(),true);
+    $dataMgr->setCourseFromID(new CourseID($params['courseID']));
+    $assignment = $dataMgr->getAssignment(new AssignmentID($params['assignmentID']));
+    make_peer_review($assignment, $params);
+});
+
+
+
+
```
Comment (madhavsuresh) [inline api/index.php]: nah

File: api/api_lib.php [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+    $db->commit();
+    }
+    catch (Exception $e){
+       echo $e->getMessage();
+    }
+}
+
+
+
+function createAssignment($assignment_params){
```
Comment (madhavsuresh) [inline api/api_lib.php]: mixing camel case and underscores

File: api/api_lib.php [hunk 6]
```diff
@@ -0,0 +1,131 @@
+<?php
+require_once("../inc/common.php");
+require_once("../peerreview/inc/common.php");
+require_once("../inc/datamanager.php");
+require_once("../peerreview/inc/datamanagers/pdoassignmentdatamanager.php");
+require_once("../config.php");
+
+function mockSubmissions($courseID, $assignmentID){
```
Comment (madhavsuresh) [inline api/api_lib.php]: camelCase

File: api/index.php [hunk 7]
```diff
@@ -125,6 +127,98 @@ function decode_json_throw_errors($inputString) {
...
+    $params = json_decode($params, true);
+    $dataMgr->setCourseFromID(new CourseID($params['courseID']));
+    $assignment = createAssignment($params); 
+});
+
+$app->post('/assignment/update', function( Request $request, Response $response) use ($dataMgr){
+    $json_params = $request->getBody();
+    $params = json_decode($json_params,true); 
+    $dataMgr->setCourseFromID(new CourseID($params['courseID']));
+    $assignment = createAssignment($params);
```
Comment (madhavsuresh) [inline api/index.php]: need to rename this function

**Comment by madhavsuresh** [comment]
there are some changes that need to be made, comments are inline, will accept for now

lgtm

**Comment by nathanshelly** [comment]
lgtm

**Comment by madhavsuresh** [comment]
need to rename your endpoints, and add json schemas

**Pull request merged**